### ISSUE TITLE
bug on deleting a ressource

### ISSUE THREAD

**Issue by yblatti** [easySDI/easySDI#707]
**Original author :** Portier Thomas

I go to my ressource list and I edit a metadata
I go back on my list with the "return" button
I try to delete a metadata and i got an error 500

**Fixed versions :** 4.2.0

**Comment by yblatti** [comment]
**Original author :** Magoni Bruno

Error HTTP 500 appears only the first time we want to delete a resource; trying to delete again same resource displays correctly the confirmation modal box!

**Comment by yblatti** [comment]
**Original author :** Battaglia Marc

unable to reproduce...

**Comment by yblatti** [comment]
**Original author :** Magoni Bruno

Issue rejected as impossible to reproduce...

**Comment by yblatti** [comment]
**Original author :** Portier Thomas

I confirm I can reproduce it.

More widely, there is a problem with links between pages of the resource panel in front-end.

My resource panel has this url : http://www.tigeo.fr/mes-ressources

If I edit a metadata, the url is switched to : http://www.tigeo.fr/component/easysdi_catalog/?view=metadata&layout=edit

It should be more like : http://www.tigeo.fr/mes-ressources/metadata/edit

Same thing appears when I want to manage visualisation : http://www.tigeo.fr/component/easysdi_map/?view=visualization&layout=edit

It should be more like : http://www.tigeo.fr/mes-ressources/visualization/edit

This probleme is caused by existence of several components (core, catalog) ?

**Comment by yblatti** [comment]
**Original author :** Mérour Xavier

Seems to be a general design problem. The point will be first point out to the PSCS (this friday)
It should be then discussed in Technical Committee as soom as possible to see how to solve this issue.

**Comment by yblatti** [comment]
**Original author :** Battaglia Marc

Ok, I reproduced the bug after editing metadata. 

The problem was the use of: 
```

 <a href="#" onclick="showDeleteModal('/resources/version/remove/170',170);">
```

instead of
```

<a href="#" onclick="showDeleteModal('/resources/version/remove/170',170); return false;">
```


Without the "return false", the page is redirected to the current url followed by #. According to the current url, this causes an error 500.

The problem is the same for visualization.

We added return false after each method used in an onClick.

**Comment by yblatti** [comment]
**Original author :** Mérour Xavier

Hello Marc,

If it solves the problem, I am happy :-)
Really thought it was deeper...

Thanks !

**Comment by yblatti** [comment]
**Original author :** Mérour Xavier

Question : should not this issue put in the next release since 4.2.0 Beta 1 is already out ?

**Comment by yblatti** [comment]
**Original author :** Magoni Bruno

Fix has been already packaged in 4.2.0 Beta 1 :-)
